gwpy/timeseries/timeseries.py

```diff
@@ -1290,6 +1290,41 @@ def rms(self, stride=1):
         return self.__class__(data, channel=self.channel, t0=self.t0,
                               name=name, sample_rate=(1/float(stride)))
 
+    def demod(self, f, stride=1):
+        """Compute the average magnitude and phase of this `TimeSeries`
+           once per stride at a given frequency.
+
+        Parameters
+        ----------
+        f : `float`
+            frequency (Hz) at which to demodulate the signal
+
+        stride : `float`
+            stride (seconds) between calculations
+
+        Returns
+        -------
+        demod : `TimeSeries`
+            a new `TimeSeries` containing magnitude and phase trends
+            (stored as a complex number) with dt=stride
+        """
```

Comment:
@alurban, can you add an example to the bottom of this docstring? See https://github.com/alurban/gwpy/blob/ea10c8076db5bfcabc1c628ac109c0b70c879c7b/gwpy/timeseries/timeseries.py#L1027 for an example of that.